MatrixSDK/Crypto/MXOlmDevice.h

```diff
@@ -178,35 +180,32 @@ Determine if an incoming messages is a prekey message matching an existing sessi
 /**
  Generate a new outbound group session.
 
- @return the session id for the outbound session.
+ @param roomId Identifer of the room.
+ @return the newly created outbound group session.
  */
-- (NSString*)createOutboundGroupSession;
+- (OLMOutboundGroupSession *)createOutboundGroupSessionForRoomWithId:(NSString *)roomId;
 
 /**
- Get the current session key of  an outbound group session.
+ Store a outbound group session with session ID for room with ID.
 
- @param sessionId the id of the outbound group session.
- @return the base64-encoded secret key.
+ @param roomId Identifer of the room.
  */
-- (NSString*)sessionKeyForOutboundGroupSession:(NSString*)sessionId;
+- (void)storeOutboundGroupSession:(OLMOutboundGroupSession *)session forRoomWithId:(NSString *)roomId;
 
 /**
- Get the current message index of an outbound group session.
+ Retrieve a outbound group session Info for a specific room.
 
- @param sessionId the id of the outbound group session.
- @return the current chain index.
+ @param roomId Identifer of the room.
+ @return the session info instance for the outbound session. Nil if not found in store.
  */
-- (NSUInteger)messageIndexForOutboundGroupSession:(NSString*)sessionId;
+- (MXOutboundSessionInfo *)outboundGroupSessionInfoForRoom:(NSString *)roomId;
 
 /**
- Encrypt an outgoing message with an outbound group session.
-
- @param sessionId the id of the outbound group session.
- @param payloadString the payload to be encrypted and sent.
- @return ciphertext
+ Remove the outbound group session for a specific room from the cache and store.
+ 
+ @param roomId Identifer of the room.
  */
-- (NSString*)encryptGroupMessage:(NSString*)sessionId payloadString:(NSString*)payloadString;
-
+- (void)discardOutboundGroupSessionForRoom:(NSString *)roomId;
```

Comment:
```suggestion
- (void)discardOutboundGroupSessionForRoomWithRoomId:(NSString *)roomId;
```

To have the same shape as the other methods in this API set.